File: test-suite/src/test/java/io/micronaut/docs/config/builder/EngineFactory.java
package io.micronaut.docs.config.builder;

// tag::imports[]
import io.micronaut.context.annotation.Factory;

import javax.inject.Singleton;
// end::imports[]

// tag::class[]
@Factory
class EngineFactory {

    @Singleton
    EngineImpl buildEngine(EngineConfig engineConfig) {
        return engineConfig.builder.build(engineConfig.crankShaft, engineConfig.getSparkPlug());
    }
}
// end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/config/builder/EngineFactory.kt
package io.micronaut.docs.config.builder

// tag::imports[]
import io.micronaut.context.annotation.Factory
import javax.inject.Singleton
// end::imports[]

// tag::class[]
@Factory
internal class EngineFactory {

    @Singleton
    fun buildEngine(engineConfig: EngineConfig): EngineImpl {
        return engineConfig.builder.build(engineConfig.crankShaft, engineConfig.sparkPlug)
    }
}
// end::class[]